subsys/nrf_security/src/drivers/cracen/cracen_sw/include/cracen_sw_aead.h

```diff
@@ -0,0 +1,241 @@
+					size_t plaintext_length);
+
+/**
+ * @brief Update software AEAD operation with additional data.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[in] input Additional data.
+ * @param[in] input_length Length of additional data.
+ *
+ * @retval PSA_SUCCESS Additional data processed successfully.
+ * @retval PSA_ERROR_NOT_SUPPORTED Unsupported algorithm.
+ * @retval PSA_ERROR_BAD_STATE AD already finalized or operation not initialized.
+ */
+psa_status_t cracen_aead_update_ad(cracen_aead_operation_t *operation, const uint8_t *input,
+				      size_t input_length);
+
+/**
+ * @brief Update software AEAD operation with plaintext/ciphertext.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[in] input Input data (plaintext for encrypt, ciphertext for decrypt).
+ * @param[in] input_length Length of input data.
+ * @param[out] output Output buffer.
+ * @param[in] output_size Size of output buffer.
+ * @param[out] output_length Pointer to store actual output length.
+ *
+ * @retval PSA_SUCCESS Data processed successfully.
+ * @retval PSA_ERROR_NOT_SUPPORTED Unsupported algorithm.
+ * @retval PSA_ERROR_BUFFER_TOO_SMALL Output buffer too small.
+ * @retval PSA_ERROR_BAD_STATE Operation not initialized.
+ */
+psa_status_t cracen_aead_update(cracen_aead_operation_t *operation, const uint8_t *input,
+				   size_t input_length, uint8_t *output, size_t output_size,
+				   size_t *output_length);
+
+/**
+ * @brief Finish software AEAD encryption and generate tag.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[out] ciphertext Final ciphertext output buffer.
+ * @param[in] ciphertext_size Size of ciphertext buffer.
+ * @param[out] ciphertext_length Pointer to store actual ciphertext length.
+ * @param[out] tag Tag output buffer.
+ * @param[in] tag_size Size of tag buffer.
+ * @param[out] tag_length Pointer to store actual tag length.
+ *
+ * @retval PSA_SUCCESS Encryption finished successfully, tag generated.
+ * @retval PSA_ERROR_NOT_SUPPORTED Unsupported algorithm.
+ * @retval PSA_ERROR_BUFFER_TOO_SMALL Tag buffer too small.
+ * @retval PSA_ERROR_BAD_STATE Operation not initialized.
+ */
+psa_status_t cracen_aead_finish(cracen_aead_operation_t *operation, uint8_t *ciphertext,
+				   size_t ciphertext_size, size_t *ciphertext_length, uint8_t *tag,
+				   size_t tag_size, size_t *tag_length);
+
+/**
+ * @brief Finish software AEAD decryption and verify tag.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[out] plaintext Final plaintext output buffer.
+ * @param[in] plaintext_size Size of plaintext buffer.
+ * @param[out] plaintext_length Pointer to store actual plaintext length.
+ * @param[in] tag Tag to verify.
+ * @param[in] tag_length Length of the tag.
+ *
+ * @retval PSA_SUCCESS Decryption finished successfully, tag verified.
```

Comment:
```suggestion
 * @retval PSA_SUCCESS Decryption finished successfully; tag verified.
```
Or `:`, as per above.